pkg/manifest/simple.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package manifest
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/ethersphere/bee/pkg/file"
+	"github.com/ethersphere/bee/pkg/file/joiner"
+	"github.com/ethersphere/bee/pkg/file/splitter"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/manifest/simple"
+)
+
+const (
+	// ManifestSimpleContentType represents content type used for noting that
+	// specific file should be processed as 'simple' manifest
+	ManifestSimpleContentType = "application/bzz-manifest-simple+json"
+)
+
+type simpleManifest struct {
+	manifest simple.Manifest
+
+	ctx       context.Context
```

Comment:
i am not entirely sure this is desirable to have on the manifest level. you might create a manifest and pass it on to some other calling context to handle and that way you lose control over cancellations in deeper calls in the cascade

Reply:
Personally I never saw manifest as something that was suppose to be shared in any special way around the code, apart from few simple operations with it. Until now there were not usages of the context in regards with manifest.

As far as the context goes, it is only ever used to save or load manifest.
Load will happen during actual constructor, while the save might be separated, however it could not devise a way to initialize manifest at single point, add stuff from multiple context-es, and save in some completely other one.

It will be easy to change this anyway.